### ISSUE TITLE
Crash when running as root on arm64 Ubuntu 20.04

### ISSUE THREAD

**Issue by kolbe** [pingcap/tidb-insight#53]
See https://github.com/pingcap/tiup/issues/1396 for full information.

```
This is a problem with https://github.com/pingcap/tidb-insight, in fact. It crashes on this platform when run under sudo but not when run as a regular user.

$ sudo /tmp/tiup/bin/insight
unexpected fault address 0xffffa065e011
fatal error: fault
[signal SIGBUS: bus error code=0x1 addr=0xffffa065e011 pc=0x4038d0]

goroutine 1 [running]:
runtime.throw(0x613286, 0x5)
        runtime/panic.go:774 +0x54 fp=0x40000c1850 sp=0x40000c1820 pc=0x42c01c
runtime.sigpanic()
        runtime/signal_unix.go:391 +0x434 fp=0x40000c1880 sp=0x40000c1850 pc=0x440b4c
memeqbody(0xffffa065e010, 0x40000c18c0, 0x5, 0x40000c1800, 0x28, 0x5f494d445f, 0x40000c1a28, 0x50c534, 0xffffa065e000, 0x1f, ...)
        internal/bytealg/equal_arm64.s:112 +0xd8 fp=0x40000c1890 sp=0x40000c1890 pc=0x4038d0
bytes.Equal(...)
        bytes/bytes.go:20
github.com/AstroProfundis/sysinfo.epsValid(0xffffa065e000, 0x1f, 0x1f, 0x1f)
        github.com/AstroProfundis/sysinfo@v0.0.0-20210201035811-eb96b87c86b3/memory.go:53 +0x80 fp=0x40000c18d0 sp=0x40000c1890 pc=0x50bc18
github.com/AstroProfundis/sysinfo.getStructureTableAddressEFI(0x40000a8140, 0x0, 0x0, 0x0, 0x0)
        github.com/AstroProfundis/sysinfo@v0.0.0-20210201035811-eb96b87c86b3/memory.go:85 +0x274 fp=0x40000c1a30 sp=0x40000c18d0 pc=0x50bf3c
github.com/AstroProfundis/sysinfo.getStructureTable(0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/AstroProfundis/sysinfo@v0.0.0-20210201035811-eb96b87c86b3/memory.go:138 +0x10c fp=0x40000c1b00 sp=0x40000c1a30 pc=0x50c534
github.com/AstroProfundis/sysinfo.(*SysInfo).getMemoryInfo(0x40000f60b0)
        github.com/AstroProfundis/sysinfo@v0.0.0-20210201035811-eb96b87c86b3/memory.go:164 +0x54 fp=0x40000c1de0 sp=0x40000c1b00 pc=0x50c72c
github.com/AstroProfundis/sysinfo.(*SysInfo).GetSysInfo(0x40000f60b0)
        github.com/AstroProfundis/sysinfo@v0.0.0-20210201035811-eb96b87c86b3/sysinfo.go:33 +0x58 fp=0x40000c1e20 sp=0x40000c1de0 pc=0x50fec0
github.com/pingcap/tidb-insight/collector/insight.(*InsightInfo).GetInfo(0x40000f6000, 0x0, 0x0, 0x0)
        github.com/pingcap/tidb-insight@/collector/insight/insight.go:62 +0xd4 fp=0x40000c1e90 sp=0x40000c1e20 pc=0x51d78c
main.main()
        github.com/pingcap/tidb-insight@/collector/collector.go:38 +0x70 fp=0x40000c1f50 sp=0x40000c1e90 pc=0x522688
runtime.main()
        runtime/proc.go:203 +0x23c fp=0x40000c1fd0 sp=0x40000c1f50 pc=0x42d8f4
runtime.goexit()
        runtime/asm_arm64.s:1128 +0x4 fp=0x40000c1fd0 sp=0x40000c1fd0 pc=0x456fcc
$ echo $?
2
```

```
$ /tmp/tiup/bin/insight
{
  "meta": {
    "timestamp": "2021-06-03T04:58:41.038860816Z",
    "uptime": 2394.15,
    "idle_time": 19108.74,
    "sysinfo_ver": "0.9.3",
    "git_branch": "heads/v0.3.2",
    "git_commit": "v0.3.2",
    "go_version": "go1.13.10 linux/arm64",
    "tidb": null,
    "tikv": null,
    "pd": null
  },
  "sysinfo": {
    "node": {
      "hostname": "ip-10-0-2-206",
      "machineid": "eb6361e1c42d415abd9a8fe62a839fb4",
      "timezone": "Etc/UTC"
    },
    "os": {
      "name": "Ubuntu 20.04.2 LTS",
      "vendor": "ubuntu",
      "version": "20.04",
      "release": "20.04.2"
    },
    "kernel": {
      "release": "5.4.0-1045-aws",
      "version": "#47-Ubuntu SMP Tue Apr 13 07:04:23 UTC 2021",
      "architecture": "aarch64"
    },
    "product": {
      "name": "t4g.2xlarge",
      "vendor": "Amazon EC2"
    },
    "board": {
      "vendor": "Amazon EC2",
      "assettag": "i-065d6b920d7be7ecc"
    },
    "chassis": {
      "type": 1,
      "vendor": "Amazon EC2",
      "assettag": "Amazon EC2"
    },
    "bios": {
      "vendor": "Amazon EC2",
      "version": "1.0",
      "date": "11/1/2018"
    },
    "cpu": {
      "threads": 8
    },
    "memory": {
      "size": 31717
    },
    "storage": [
      {
        "name": "nvme0n1",
        "model": "Amazon Elastic Block Store",
        "serial": "vol0129d4919ce22fda4",
        "size": 68
      }
    ],
    "network": [
      {
        "name": "ens5",
        "driver": "ena",
        "macaddress": "06:24:f2:b6:7f:49",
        "ipaddress": [
          "10.0.2.206/24",
          "fe80::424:f2ff:feb6:7f49/64"
        ],
        "mtu": 9001
      }
    ]
  },
  "ntp": {
    "version": "exec: \"ntpq\": executable file not found in $PATH"
  },
  "partitions": [
    {
      "name": "nvme0n1",
      "mount": {},
      "sectors": 134217728,
      "size": 68719476736,
      "subdev": [
        {
          "name": "nvme0n1p1",
          "partition": true,
          "mount": {},
          "uuid": "cb9103d6-ca5d-4862-9bc7-41fa9d819898",
          "sectors": 134010847,
          "size": 68613553664
        },
        {
          "name": "nvme0n1p15",
          "partition": true,
          "mount": {
            "mount_point": "/boot/efi",
            "filesystem": "vfat",
            "mount_options": "rw,relatime,fmask=0077,dmask=0077,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro"
          },
          "uuid": "7CB3-16E6",
          "sectors": 202753,
          "size": 103809536
        }
      ],
      "rotational": "0"
    },
    {
      "name": "md0",
      "mount": {},
      "rotational": "1"
    }
  ],
  "epoll_exclusive": true
}
$ echo $?
0
```

**Comment by 9547** [comment]
Can't reproduce it with RaspberryPi 3B arm32 on Debian:

```bash
root@home-blackpi:/tmp # lscpu
Architecture:          armv7l
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    1
Core(s) per socket:    4
Socket(s):             1
Model:                 4
Model name:            ARMv7 Processor rev 4 (v7l)
CPU max MHz:           1400.0000
CPU min MHz:           600.0000
BogoMIPS:              89.60
Flags:                 half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32

root@home-blackpi:/tmp # hostnamectl
   Static hostname: home-blackpi
         Icon name: computer
        Machine ID: 51e65fdbaf...895712ad80f
           Boot ID: 94b6f8e07a7...1ebc01dcb45a7
  Operating System: Raspbian GNU/Linux 9 (stretch)
            Kernel: Linux 4.14.50-v7+
      Architecture: arm

root@home-blackpi:/tmp # lsb_release -a
No LSB modules are available.
Distributor ID: Raspbian
Description:    Raspbian GNU/Linux 9.4 (stretch)
Release:        9.4
Codename:       stretch
```

**Comment by AstroProfundis** [comment]
This is the same as https://github.com/pingcap/tiup/issues/1337 I can't reproduce it on my environments. However, some early debugging indicates this might be due to some specific memory layout or other hardware setups. The panic itself is in `sysinfo` package and I don't yet have any clue how to fix it.

**Comment by kolbe** [comment]
https://github.com/zcalusic/sysinfo/issues/26

**Comment by AstroProfundis** [comment]
> [zcalusic/sysinfo#26](https://github.com/zcalusic/sysinfo/issues/26)

Indeed, we now have access to an server with this issue for a while, we'll try to see if there could be any workaround.

**Comment by lucklove** [comment]
@kolbe I have created a [fix](https://github.com/zcalusic/sysinfo/pull/27) and the upstream merged it, can you test sysinfo with the newest master code again?

**Comment by kolbe** [comment]
@lucklove it should be trivial for someone in R&D to confirm that this tool and by extension `tiup check` can work on the AWS ec2 instance type where I saw a problem. I am not very interested in trying to build this component and verify that it works correctly with TiUP in this environment.

**Comment by lucklove** [comment]
@kolbe OK, I'll test it with TiUP then. Thanks for your issue.

**Comment by AstroProfundis** [comment]
It has been fixed in upstream by @lucklove and `insight` on the `tiup` branch should be working now. We'll release it later on, as we have a potential bug on TiUP side that may not working correctly with latest `insight`, and I'm still waiting (another) upstream to merge my fix.

**Comment by AstroProfundis** [comment]
This should already been fixed when insight `v0.4.0` released.
